app/controllers/api/vms_controller.rb

```diff
@@ -263,6 +173,19 @@ def set_miq_server_resource(type, id, data)
 
     private
 
+    def exec_vm_resource_action(verb, action, method, type, id = nil)
```

Comment:
Does there need to be a default argument for id?

Also, I'd prefer no abbreviations in these method names.

5 arguments is a lot of arguments. Did you consider using kwargs (or a hash) to better convey what they do from the calling end?